docs/chapter4-payload-handoff-format.rst

```diff
@@ -32,6 +32,104 @@ architecture, either 32 or 64 bits.
+
+   ================= =========== ============ ======================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================
+   Property          Usage       Value Type   Definition
+   ================= =========== ============ ======================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================
+   compatible        R           string       "upl"
+   boot-mode         O           string list  For supporting different Payload boot flows. These influence how the payload runs. Note that this is not intended to support multiple Payloads.
+
+                                              “normal” means booting with full functionality, from power off.
+                                              “fast” means fast boot with no configuration change or minimal functionality.
+                                              “full", "diag” means booting with full functionality plus diagnostics functions.
+                                              “default” means booting with “reset to default” flow (for example, boot from RTC battery power failure).
+                                              “s4” means resuming from ACPI S4 state (resume from disk).
+                                              “factory” means booting with manufacturing mode.
+                                              “s3” means resuming from ACPI S3 state (resume from memory).
+
+                                              Payload is not required to support all of the boot modes. For boot modes those are not supported by Payload, Payload can just ignore them without any specific actions.
+   addr-width        O           u32          52 means host address width is 52 bits. 46 means address width is 46 bits.
+                                              This is mainly used by Payloads when initializing page tables. This can eliminate directly SoC accessing from payload for retrieving such information.
+   pci-enum-done     O           Boolean      If this property exists, payload can assume that Platform Init has done full PCI enumeration, resource assignment, and PCI Oprom loading, and those do not have to be done again by payload.
```

Comment:
Do we need mention that if pci-enum-done is done by Platform Init, where the payload could get the assigned pci resource info (to avoid pci enumeration in payload)?

Reply:
Make sense. @ChaselChiu do you know where could payload get the pci enum info?

Reply:
We would like to connect pci-rb node to this pci-enum-done property and clarify the payload behavior like below:

When pci-enum-done present, payload will skip PCI resource assignment and it will use pci-rb node information to create internal resource map to ensure no conflict usage of those in-used PCI resources during payload execution.

When pci-enum-done not present, payload may do the PCI enumeration to assign resources basing on pci-rb node information, and will also create resource map too.

Reply:
Done.